src/common/darktable.c

```diff
@@ -756,9 +756,27 @@ int dt_init(int argc, char *argv[], const gboolean init_gui, const gboolean load
 #ifdef _OPENMP
   omp_set_num_threads(darktable.num_openmp_threads);
 #endif
-  dt_loc_init_datadir(datadir_from_command);
-  dt_loc_init_plugindir(moduledir_from_command);
-  dt_loc_init_localedir(localedir_from_command);
+
+  // get the application directory from the first argument containing the full path to the executable
```

Comment:
This is actually incorrect - argv[0] contains full path only if you start the program by using the full path. E.g. if you start it with just darktable, "darktable" is exactly what you get in argv[0].

Reply:
Indeed, the comment was wrong (fixed).

Reply:
If this comment is wrong, then how does it work when darktable is started with just "darktable"? Have you tested this (most common BTW) case? Because you have removed hard-coded absolute paths from the code.

Reply:
On my linux box I have tested 

- ./darktable
- ./some/path/to/darktable
- ~/some/path/to/darktable
- ~username/some/path/to/darktable

and that all works fine. Now you point me on a usecase I did not test yet 

- darktable (while having the location in the PATH)
And this does not work as in this case application_directory resolves to NULL. I will have to fix that using /proc/self/exe on linux and GetModuleHandle on windows.

Reply:
@parafin https://stackoverflow.com/a/1024937/1399018 this pretty nicely covers the current state and the upcoming problems. Any thoughts on this?

Reply:
I would say "good luck";) For macOS bundle it is solved in dt_osx_get_bundle_res_path.

Reply:
Why reinventing the wheel when there are libraries that take over that task. I decided for [whereami](https://github.com/gpakosz/whereami).

Reply:
Should be solved now. I have tested various versions and it works for now. No Idea yet about other Os. But the lib states:

    Supported platforms:

    Windows
    Linux
    Mac
    iOS
    Android
    QNX Neutrino
    FreeBSD
    NetBSD
    DragonFly BSD
    SunOS


I also don't know if the license adds any burden to our project. There is a MIT and a WTFPLv2  license available. @TurboGit from the license side the library works for us, right?